packages/vite-plugin-svelte/src/utils/log.ts

```diff
@@ -104,13 +104,17 @@ export function logCompilerWarnings(warnings: Warning[], options: ResolvedOption
 	const warn = isBuild ? warnBuild : warnDev;
 	const notIgnoredWarnings = warnings?.filter((w) => !ignoreCompilerWarning(w, isBuild, emitCss));
 	const extraWarnings = buildExtraWarnings(warnings, isBuild);
-	[...notIgnoredWarnings, ...extraWarnings].forEach((warning) => {
+	const warningsToSend = [...notIgnoredWarnings, ...extraWarnings];
+	warningsToSend.forEach((warning) => {
 		if (onwarn) {
 			onwarn(warning, warn);
 		} else {
 			warn(warning);
 		}
 	});
+	if (!isBuild && options.experimental?.sendWarningsToBrowser) {
+		options.server?.ws?.send('svelte:warnings', warningsToSend);
+	}
```

Comment:
Should we send the warning in the `warnDev` function instead? That way user-suppressed warning don't make it to the browser. 

Maybe it makes sense to send the warning one-by-one too with this way.

Reply:
The receiver needs to be able to know what all the warnings for one file are at a single point in time to be able to evict outdated warnings. If we send them one-by-one it could a lot of messages too.

Good point about user exclustions tough. Any ideas for alternate solutions?